sklearn/tree/tests/test_tree.py

```diff
@@ -2406,3 +2407,25 @@ def test_splitter_serializable(Splitter):
     splitter_back = pickle.loads(splitter_serialize)
     assert splitter_back.max_features == max_features
     assert isinstance(splitter_back, Splitter)
+
+
+def test_tree_deserialization_from_read_only_buffer():
+    # Non-regression test for the issue:
+    # https://github.com/scikit-learn/scikit-learn/issues/25584
+    # This test helps to ensure that the Tree module is able to
+    # successfully deserialize data from read only buffers.
+    X, y = fetch_california_housing(return_X_y=True, as_frame=True)
```

Comment:
We try not to use `fetch_*` functions in test because they require the network and some package managers will prevent network requests in tests.

Going from https://github.com/scikit-learn/scikit-learn/issues/25584#issuecomment-1425795214 here is a simple test to trigger the error:

```python
def test_tree_deserialization_from_read_only_buffer(tmp_path):
    pickle_path = str(tmp_path / "clf.joblib")
    tree = DecisionTreeClassifier(random_state=0)
    tree.fit(X_small, y_small)

    joblib.dump(tree, pickle_path)
    joblib.load(pickle_path, mmap_mode="r")
```